src/stac_api_benchmark/query.py

```diff
@@ -154,6 +160,10 @@ async def search(
             config.logger.error(f"{search_id}: APIError: {e}")
             time = perf_counter() - t_start
             return -1, time
+        except TimeoutError as e:
+            config.logger.error(f"{search_id}: TimeoutError ({config.timeout}s): {e}")
+            time = perf_counter() - t_start
+            return -1, time
```

Comment:
I hadn't noticed before that this is a tuple return -- what's the semantics of the tuple? Does the first value just map to `sys.exit` return codes?

Reply:
it's the number of results -- probably time to use a dataclass instead. This is one of things I want to refactor -- ideally, this would be a https://github.com/dry-python/returns#result-container

Reply:
I <3 `Result` ovi :-)